GitTfs/Core/Changes/Git/RenameEdit.cs

```diff
@@ -6,24 +7,100 @@ public class RenameEdit : IGitChangedFile
         public string Path { get; private set; }
         public string PathTo { get; private set; }
         public string NewSha { get; private set; }
+        public string OldSha { get; private set; }
         public string Score { get; private set; }
         public IGitRepository _repository { get; private set; }
 
         public RenameEdit(IGitRepository repository, GitChangeInfo changeInfo)
         {
             _repository = repository;
             NewSha = changeInfo.newSha;
+            OldSha = changeInfo.oldSha;
             Path = changeInfo.path;
             PathTo = changeInfo.pathTo;
             Score = changeInfo.score;
         }
 
         public void Apply(ITfsWorkspaceModifier workspace)
         {
-            workspace.Edit(Path);
-            workspace.Rename(Path, PathTo, Score);
-            var workspaceFile = workspace.GetLocalPath(PathTo);
-            _repository.CopyBlob(NewSha, workspaceFile);
+            if (Path.ToLower() != PathTo.ToLower())
+            {
+                var workspaceFile = workspace.GetLocalPath(PathTo);
+                workspace.Edit(Path);
+                workspace.Rename(Path, PathTo, Score);
+                _repository.CopyBlob(NewSha, workspaceFile);
+            }
+            else
+            {
+                //This change can not be mapped to TFS, since TFS is case insensitive
+                var workspaceFile = workspace.GetLocalPath(Path);
```

Comment:
I was under the impression, that the file name is used for calculating the hash.
Since this is not the case, the file comparison is unnecessary.